lib/node-utils/index.ts

```diff
@@ -76,6 +81,23 @@ export function findClosestCallExpressionNode(
   return findClosestCallExpressionNode(node.parent, shouldRestrictInnerScope);
 }
 
+export function findClosestVariableDeclaratorNode(
+  node: TSESTree.Node | undefined
+): TSESTree.VariableDeclarator | null {
+  if (!node) {
+    return null;
+  }
+
+  if (ASTUtils.isVariableDeclarator(node)) {
+    return node;
+  }
+
+  return findClosestVariableDeclaratorNode(node.parent);
+}
```

Comment:
I'm pretty sure this could be refactored in a similar way to `isNodeOfType` so it would be possible to find the closest node of any kind, but I don't think this is the PR to do so.